### ISSUE TITLE
odo service create --dry-run is not working when used with parameters

### ISSUE THREAD

**Issue by kadel** [redhat-developer/odo#4913]
/kind bug

## How did you run odo exactly?
```

odo service create postgresql-operator.v0.1.1/Database -h database -p size=1 --dry-run
```

## Actual behavior
```
▶ odo service create postgresql-operator.v0.1.1/Database database -p size=1 --dry-run
 ✗  couldn't find metadata.name in the yaml; provide a name for the service
```

## Expected behavior

```
▶ odo service create postgresql-operator.v0.1.1/Database database  -p size=1 --dry-run
apiVersion: postgresql.dev4devs.com/v1alpha1
kind: Database
metadata:
  name: database
spec:
  size: 1
```

**Comment by dharmit** [comment]
Isn't this same as #4813? 🤔

**Comment by dharmit** [comment]
If this doesn't get fixed by PR for #4813, we work on this after that.

**Comment by kadel** [comment]
this commands no longer exists in odo v3
/close
